src/verbosity.jl

```diff
@@ -0,0 +1,89 @@
+function restore_callsite_source_position!(expr, src)
+    # used to fix the logging source file + line
+    # since we're lowering our verbose logging macros to the
+    # Logging.jl macros; otherwise, they would always report this (verbosity.jl)
+    # file as the logging callsite
+    expr.args[1].args[2] = src
+    return expr
+end
+
+vlogmacrodocs = """
+    @debugv N msg args...
+    @infov N msg args...
+    @warnv N msg args...
+    @errorv N msg args...
+
+"Verbose" logging macros. Drop in replacements of standard logging macros, but an
+additional verbosity level `N` is passed to indicate differing verbosity levels
+for a given log level. The verbosity argument is subtracted from the base log level when passed down
+to the core logging logic, so `@debugv 1 msg` will essentially call `@logmsg Debug-1 msg`.
+
+An `LoggingExtras.LevelOverrideLogger`can then be used to filter on the `level` argument.
+For convenience, the
+[`LoggintExtras.with(f; level, verbosity)`](@ref) function is provided to temporarily
+wrap the current logger with a log level and verbosity subtracted to filter while `f` is executed.
+"""
+
+macro debugv(verbosity::Int, msg, exs...)
+    return restore_callsite_source_position!(
+        esc(:($Base.@logmsg (Logging.Debug - $verbosity) $msg $(exs...))),
+        __source__,
+    )
+end
+
+macro infov(verbosity::Int, msg, exs...)
+    return restore_callsite_source_position!(
+        esc(:($Base.@logmsg (Logging.Info - $verbosity) $msg $(exs...))),
+        __source__,
+    )
+end
+
+macro warnv(verbosity::Int, msg, exs...)
+    return restore_callsite_source_position!(
+        esc(:($Base.@logmsg (Logging.Warn - $verbosity) $msg $(exs...))),
+        __source__,
+    )
+end
+
+macro errorv(verbosity::Int, msg, exs...)
+    return restore_callsite_source_position!(
+        esc(:($Base.@logmsg (Logging.Error - $verbosity) $msg $(exs...))),
+        __source__,
+    )
+end
+
+macro logmsgv(verbosity::Int, level, msg, exs...)
```

Comment:
Rather than using the macros to add the docs after the fact, maybe we can just
```suggestion
"$vlogmacrodocs"
macro logmsgv(verbosity::Int, level, msg, exs...)
```

Reply:
Pushed an update to fix all the macros like this